hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/SCMDBInsightEndPoint.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.hadoop.hdds.scm.container.ContainerID;
+import org.apache.hadoop.hdds.scm.container.ContainerInfo;
+import org.apache.hadoop.hdds.utils.db.DBStore;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.TableIterator;
+import org.apache.hadoop.ozone.recon.api.types.ContainerBlocksInfoWrapper;
+import org.apache.hadoop.ozone.recon.api.types.DeletedContainerInfo;
+import org.apache.hadoop.ozone.recon.scm.ReconContainerManager;
+import org.apache.hadoop.ozone.recon.scm.ReconStorageContainerManagerFacade;
+
+import javax.inject.Inject;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.hadoop.hdds.scm.metadata.SCMDBDefinition.DELETED_BLOCKS;
+import static org.apache.hadoop.ozone.recon.ReconConstants.DEFAULT_FETCH_COUNT;
+import static org.apache.hadoop.ozone.recon.ReconConstants.PREV_DELETED_BLOCKS_TRANSACTION_ID_DEFAULT_VALUE;
+import static org.apache.hadoop.ozone.recon.ReconConstants.PREV_CONTAINER_ID_DEFAULT_VALUE;
+import static org.apache.hadoop.ozone.recon.ReconConstants.RECON_QUERY_LIMIT;
+import static org.apache.hadoop.ozone.recon.ReconConstants.RECON_QUERY_PREVKEY;
+
+/**
+ * Endpoint to get following information about containers and blocks under SCM
+ * DB Insight page of Recon.
+ * 1. Number of DELETED containers.
+ * 2. Number of blocks pending deletion.
+ *     - Blocks pending deletion for open/closing containers.
+ *     - Blocks pending deletion for closed containers.
+ */
+@Path("/containers")
+@Produces(MediaType.APPLICATION_JSON)
+@AdminOnly
+public class SCMDBInsightEndPoint {
+  private final DBStore scmDBStore;
+  private final ReconContainerManager containerManager;
+
+  @Inject
+  public SCMDBInsightEndPoint(ReconStorageContainerManagerFacade reconSCM) {
+    this.containerManager =
+        (ReconContainerManager) reconSCM.getContainerManager();
+    this.scmDBStore = reconSCM.getScmDBStore();
+  }
+
+  /**
+   * This API will return all DELETED containers in SCM in below JSON format.
+   * {
+   * containers: [
+   * {
+   *  containerId: 1,
+   *  state: DELETED,
+   *  pipelineId: "a10ffab6-8ed5-414a-aaf5-79890ff3e8a1",
+   *  numOfKeys: 3,
+   *  inStateSince: <stateEnterTime>
+   * },
+   * {
+   *  containerId: 2,
+   *  state: DELETED,
+   *  pipelineId: "a10ffab6-8ed5-414a-aaf5-79890ff3e8a1",
+   *  numOfKeys: 6,
+   *  inStateSince: <stateEnterTime>
+   * }
+   * ]
+   * }
+   * @param limit limits the number of deleted containers
+   * @param prevKey previous container Id to skip
+   * @return Response of delete containers.
+   */
+  @GET
+  @Path("/deletedContainers")
```

Comment:
deletedContainers this is already prefixed by containers/. So Containers again is not required , it can be deletedstate or deleted or whatever is more appropriate in lowercase only.

Reply:
ok. will rename to "/deleted"